dtool/src/dtoolbase/dtool_platform.h

```diff
@@ -65,6 +65,9 @@
 #define DTOOL_PLATFORM "android_i386"
 #endif
 
+#elif defined(__wasi__)
+#define DTOOL_PLATFORM "wasi"
+
```

Comment:
[wasi has arch tags](https://github.com/WebAssembly/wasi-libc/blob/be1704a9568e98ae073f074f7271cae68c23d161/libc-top-half/musl/src/misc/uname.c#L18-L21) just like emscripten does, so distinguish wasi_wasm64/wasi_wasm32 for future extensibility